### PR TITLE
Remove reliance on le-utils based pycountry using getlang methods.

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Removes use of methods in le-utils for retrieving languages by alpha3 and alpha2 tags
* Use internal methods for this only

## Reviewer guidance
This does clean up so we can remove utilities not used anywhere else in the ecosystem from le-utils.
In general we should defer to the langcodes and pycountry libraries to handle this sort of thing.

## Summary by CodeRabbit

- **New Features**
  - Improved language code handling for subtitles using enhanced lookup and fallback logic, increasing robustness and compatibility.
- **Bug Fixes**
  - Enhanced error messages in subtitle language compatibility tests for clearer debugging.
- **Chores**
  - Updated dependencies to include the latest language code data and removed unused packages.
- **Tests**
  - Standardized language lookup in tests for consistency.
  - Adjusted test fixtures for more accurate language mapping scenarios.
  - Added new tests to verify language code resolution and edge cases.

### CODE REVIEW

File: ricecooker/utils/youtube.py [hunk 1]
```diff
@@ -337,12 +338,16 @@ def get_language_with_alpha2_fallback(language_code):
...
+    try:
+        language = langcodes.get(language_code)
+    except LookupError:
+        return
+    if not language.is_valid():
+        return
+    # try to lookup `language` using either the default tag or the alpha3 tag
+    language_obj = languages.getlang(language.to_tag()) or languages.getlang(
+        language.to_alpha3()
+    )
```
Comment (nucleogenesis) [inline ricecooker/utils/youtube.py:350]: Will this be returning `None` if the calls to getlang() goes sideways somehow?

File: ricecooker/utils/youtube.py [hunk 2]
```diff
@@ -337,12 +338,16 @@ def get_language_with_alpha2_fallback(language_code):
     using the `getlang_by_alpha2` helper method.
     Returns either a le-utils Language object or None if both lookups fail.
     """
-    # 1. try to lookup `language` using internal representation
-    language_obj = languages.getlang(language_code)
-    # if language_obj not None, we know `language` is a valid language_id in the internal repr.
-    if language_obj is None:
-        # 2. try to match by two-letter ISO code
-        language_obj = languages.getlang_by_alpha2(language_code)
+    try:
```
Comment (nucleogenesis) [inline ricecooker/utils/youtube.py:341]: The docstring here still refers the `getlang_by_alpha2` function - maybe should be updated to match the new implementation?

**Review by nucleogenesis — COMMENTED** [review]
I gave everything a look and the changes all lgtm overall. Just a note about the docstring for the updated function.

File: ricecooker/utils/youtube.py (hunk 1 above)
Comment (rtibbles) [reply ricecooker/utils/youtube.py:350]: Yes, it returns None if no language matches the code.

**Review by akolson — APPROVED** [review]
Changes look correct to me! Thanks @rtibbles!
